addons/l10n_ro_efactura/models/account_move.py

```diff
@@ -0,0 +1,130 @@
+    )
+    l10n_ro_edi_state = fields.Selection(
+        selection=[
+            ('invoice_sending', 'Sending'),
+            ('invoice_sending_failed', 'Error'),
+            ('invoice_sent', 'Sent'),
+        ],
+        string='E-Factura Status',
+        compute='_compute_from_l10n_ro_edi_document_ids_stored',
+        store=True,
+    )
+    l10n_ro_edi_message = fields.Char(compute='_compute_from_l10n_ro_edi_document_ids')
+
+    @api.depends('l10n_ro_edi_document_ids')
+    def _compute_from_l10n_ro_edi_document_ids(self):
+        for move in self:
+            active_document = self.env['l10n_ro_edi.document'].search([('invoice_id', '=', move.id)], limit=1)
+            move.l10n_ro_edi_active_document_id = active_document
+            move.l10n_ro_edi_message = active_document.message
+
+    @api.depends('l10n_ro_edi_document_ids')
+    def _compute_from_l10n_ro_edi_document_ids_stored(self):
+        for move in self:
+            active_document = self.env['l10n_ro_edi.document'].search([('invoice_id', '=', move.id)], limit=1)
+            move.l10n_ro_edi_state = active_document.state
+
+    @api.depends('l10n_ro_edi_state')
+    def _compute_show_reset_to_draft_button(self):
+        """ Prevent user to reset move to draft when there's an
+            active sending document or an OK response has been received """
+        super()._compute_show_reset_to_draft_button()
+        for move in self:
+            if move.l10n_ro_edi_state in ('invoice_sending', 'invoice_sent'):
+                move.show_reset_to_draft_button = False
+
+    def _l10n_ro_edi_get_attachment_file_name(self):
+        self.ensure_one()
+        return f"{self.name.replace('/', '_')}.xml"
+
+    def _l10n_ro_edi_compute_errors(self, xml_data):
+        """ Compute possible errors before sending E-Factura """
+        self.ensure_one()
+        errors = []
+        if not self.company_id.l10n_ro_edi_access_token:
+            errors.append(_('Romanian access token not found. Please generate or fill it in the settings.'))
+        if not xml_data:
+            errors.append(_('CIUS-RO XML attachment not found.'))
+        return errors
+
+    def _l10n_ro_edi_send_invoice(self, xml_data):
+        """ Called by _call_web_service_after_invoice_pdf_render in account.move.send.
+            The state flow of Romanian E-Factura invoice goes as follows:
+             - Pre-check any errors from invoice's company before sending, return if found any
+             - Send to E-Factura, and based on the result:
+                 - if error -> delete all other error documents and create a new error document
+                 - if success -> delete all other sending documents and create a new sending document"""
+        self.ensure_one()
+        if errors := self.company_id._l10n_ro_edi_get_errors_pre_request():
+            self.env['l10n_ro_edi.document']._create_document_invoice_sending_failed(self, '\n'.join(errors))
```

Comment:
I think the methods creating the invoice document should be on the invoice model